test/Window/WindowHandleWin32.test.cpp

```diff
@@ -0,0 +1,143 @@
+                                                  CW_USEDEFAULT,
+                                                  CW_USEDEFAULT,
+                                                  640,
+                                                  480,
+                                                  nullptr,
+                                                  nullptr,
+                                                  classInfo.hInstance,
+                                                  nullptr);
+    REQUIRE(handle);
+    REQUIRE(!gotWmShowWindow);
+    REQUIRE(IsWindow(handle));
+
+    RECT windowRect{};
+    REQUIRE(GetWindowRect(handle, &windowRect));
+    const auto position = sf::Vector2i(sf::Vector2(windowRect.left, windowRect.top));
+
+    RECT clientRect{};
+    REQUIRE(GetClientRect(handle, &clientRect));
+    const auto size = sf::Vector2u(sf::Vector2(clientRect.right - clientRect.left, clientRect.bottom - clientRect.top));
+
+    SECTION("sf::WindowBase")
+    {
+        std::optional<sf::WindowBase> windowBase;
+
+        SECTION("WindowHandle constructor")
+        {
+            windowBase.emplace(handle);
+        }
+
+        SECTION("create(WindowHandle)")
+        {
+            windowBase.emplace().create(handle);
+        }
+
+        CHECK(windowBase->isOpen());
+        CHECK(windowBase->getPosition() == position);
+        CHECK(windowBase->getSize() == size);
+        CHECK(windowBase->getNativeHandle() == handle);
+    }
+
+    SECTION("sf::Window")
+    {
+        std::optional<sf::Window> window;
+
+        SECTION("Default context settings")
+        {
+            SECTION("WindowHandle constructor")
+            {
+                window.emplace(handle);
+            }
+
+            SECTION("create(WindowHandle)")
+            {
+                window.emplace().create(handle);
+            }
+
+            CHECK(window->getSettings().attributeFlags == sf::ContextSettings::Default);
+        }
+
+        SECTION("Custom context settings")
+        {
+            static constexpr sf::ContextSettings contextSettings{/* depthBits*/ 1, /* stencilBits */ 1, /* antiAliasingLevel */ 1};
+
+            SECTION("WindowHandle constructor")
+            {
+                window.emplace(handle, contextSettings);
+            }
+
+            SECTION("create(WindowHandle)")
+            {
+                window.emplace().create(handle, contextSettings);
+            }
+
+            CHECK(window->getSettings().depthBits >= 1);
+            CHECK(window->getSettings().stencilBits >= 1);
+            CHECK(window->getSettings().antiAliasingLevel >= 1);
+        }
+
+        CHECK(window->isOpen());
+        CHECK(window->getPosition() == position);
+        CHECK(window->getSize() == size);
+        CHECK(window->getNativeHandle() == handle);
+    }
+
+    CHECK(gotWmShowWindow);
+    CHECK(IsWindow(handle)); // The window is not destroyed
+
+    CHECK(DestroyWindow(handle));
```

Comment:
It would be interesting to destroy the window while the sfml object is still alive,
I am unsure if this is something that is supported?